File: client-mock-server/src/main/java/org/hle/clientmockserver/config/GrpcConfig.java
package org.hle.clientmockserver.config;

import io.grpc.ManagedChannel;
import io.grpc.ManagedChannelBuilder;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class GrpcConfig {
    @Value("${grpc.server.url}")
    private String grpcServerUrl;

    @Value("${grpc.server.port}")
    private int grpcServerPort;

    @Bean
    public ManagedChannel grpcChannel() {

        return ManagedChannelBuilder.forAddress(grpcServerUrl, grpcServerPort)
                .usePlaintext()
                .build();
    }
}

File: grpc-http1-proxy/grpc-http-proxy-spring/grpc-http1-proxy-wf-kt-native/src/main/kotlin/org/hle/grpchttp1wfn/config/GrpcConfig.kt
package org.hle.grpchttp1wfn.config

import io.grpc.ManagedChannel
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.grpc.client.GrpcChannelFactory

@Configuration
class GrpcConfig {
    @Bean
    fun channel(factory: GrpcChannelFactory): ManagedChannel =
        factory.createChannel("local")
}
